[ci] Builds are failing with different xharness versions

### Description

The builds on main are failing because they are trying to update xharness globally to a net8 version, while we also use a net9 version on net9 branch on same machine.. this causes the build to fail to update to a older version .

### Steps to Reproduce

_No response_

### Link to public reproduction project repository

_No response_

### Version with bug

8.0.5

### Is this a regression from previous behavior?

No, this is something new

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

_No response_

### Did you find any workaround?

_No response_

### Relevant log output

```shell
Starting: install xharness
==============================================================================
Task         : Command line
Description  : Run a command line script using Bash on Linux and macOS and cmd.exe on Windows
Version      : 2.231.0
Author       : Microsoft Corporation
Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/command-line
==============================================================================
Generating script.
Script contents:
dotnet tool update Microsoft.DotNet.XHarness.CLI --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version "8.0.0-prerelease*" -g
========================== Starting Command Output ===========================
/bin/bash --noprofile --norc /Users/builder/azdo/_work/_temp/3c268dd5-6843-49cb-a0a0-d3f8c1948cf9.sh
The requested version 8.0.0-prerelease.24060.1 is lower than existing version 9.0.0-prerelease.23606.1.
##[error]Bash exited with code '1'.
Finishing: install xharness
```

--- a/build.ps1
+++ b/build.ps1
@@ -9,11 +9,11 @@ Param(
 )
 
 # Restore Cake tool
-# & dotnet tool restore
-$tools = Get-Content ".config/dotnet-tools.json" | ConvertFrom-Json
-foreach ($tool in $tools.tools.PsObject.Properties) {
-    & dotnet tool install $tool.Name --version $tool.Value.version
-}
+& dotnet tool restore
+# $tools = Get-Content ".config/dotnet-tools.json" | ConvertFrom-Json
+# foreach ($tool in $tools.tools.PsObject.Properties) {
+#     & dotnet tool install $tool.Name --version $tool.Value.version
+# }
 
 # Build Cake arguments
 $cakeArguments = @("$Script");

--- a/eng/pipelines/common/maui-templates.yml
+++ b/eng/pipelines/common/maui-templates.yml
@@ -157,8 +157,8 @@ jobs:
         DOTNET_TOKEN: $(dotnetbuilds-internal-container-read-token)
         PRIVATE_BUILD: $(PrivateBuild)
 
-    - script: dotnet tool update Microsoft.DotNet.XHarness.CLI --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version "8.0.0-prerelease*" -g
-      displayName: install xharness
+    # - script: dotnet tool update Microsoft.DotNet.XHarness.CLI --add-source https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet-eng/nuget/v3/index.json --version "9.0.0-prerelease*" -g
+    #   displayName: install xharness
 
     - pwsh: ./build.ps1 --target=dotnet-integration-build --verbosity=diagnostic
       displayName: Build Microsoft.Maui.IntegrationTests

--- a/eng/pipelines/common/variables.yml
+++ b/eng/pipelines/common/variables.yml
@@ -6,7 +6,7 @@ variables:
 - name: DOTNET_SKIP_FIRST_TIME_EXPERIENCE
   value: true
 - name: DOTNET_VERSION
-  value: 7.0.401
+  value: 8.0.101
 - name: REQUIRED_XCODE
   value: 15.0.0
 - name: DEVICETESTS_REQUIRED_XCODE
@@ -52,7 +52,7 @@ variables:
 - name: TeamName
   value: Maui
 - name: POWERSHELL_VERSION
-  value: 7.1.3
+  value: 7.4.0
 - name: Codeql.Enabled
   value: false
 - group: Xamarin-Secrets

